### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by joshuaadickerson**
Working on making caching work better. Not yet implementing a caching library but at least taking some things from caching libraries to make it better. Also trying to reduce the code devoted to caching outside of the caching library

### CODE REVIEW

**Comment by emanuele45** [comment]
Ouch... now this is not merging...

File: sources/admin/BBC.controller.php [hunk 1]
```diff
@@ -1472,8 +1472,8 @@ public function action_setorder()
 			}
 		}
 
-		cache_put_data('parsing_smileys', null, 480);
-		cache_put_data('posting_smileys', null, 480);
+		Cache::instance()->put('parsing_smileys', null, 480);
+		Cache::instance()->put('posting_smileys', null, 480);
```
Comment (Spuds) [inline sources/admin/BBC.controller.php]: should those be ->remove ? (and at 1691 & 1766)

**Comment by Spuds** [comment]
Nice update !

File: sources/admin/BBC.controller.php (hunk 1 above)
Comment (emanuele45) [inline sources/admin/BBC.controller.php]: Then those in ManageLanguages.controller.php as well. :innocent:

**Comment by joshuaadickerson** [comment]
Okay, I finally have all tests passing and there are no conflicts. Please merge before I break it again lol

**Pull request merged**